src/v2/view-builder/views/captcha/CaptchaView.js

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+const HCAPTCHA_URL = 'https://hcaptcha.com/1/api.js?onload=onCaptchaLoaded&render=explicit';
+const RECAPTCHAV2_URL = 'https://www.google.com/recaptcha/api.js?onload=onCaptchaLoaded&render=explicit';
+
+import { View } from 'okta';
+import Enums from 'util/Enums';
+
+export default View.extend({
+  className: 'captcha-view',
+
+  initialize() {
+    if (this.options.appState.get('captcha')) {
+      this.captchaConfig = this.options.appState.get('captcha');
+      this._addCaptcha();
+    }
+  },
+
+  remove: function(){
+    // Cleanup global Captcha references
+    if (this.captchaConfig.type === 'HCAPTCHA') {
+      window.hcaptcha = undefined;
+    } else if (this.captchaConfig.type === 'RECAPTCHA_V2') {
+      window.grecaptcha = undefined;
+    }
+
+    View.prototype.remove.apply(this, arguments);
+  },
+
+  /**
+   *  Load the CAPTCHA lib dynamically (either HCAPTCHA or RECAPTCHAV2). Once loaded, trigger an event to inform
+   *  the parent form to actually render the CAPTCHA.
+  * */   
+  _addCaptcha() {
+
+    // Callback invoked when CAPTCHA is solved.
+    this.onCaptchaSolved = (token) => {
```

Comment:
looks like `onCaptchaSolved` can be local and doesn't have to bind to `this`.

Reply:
I bind it to `this` so that it's easier to unit test. I'll add a comment around this as well to be more clear.